src/main/java/com/aws/greengrass/util/platforms/windows/WindowsPlatform.java

```diff
@@ -221,6 +226,14 @@ protected void setOwner(UserPrincipal userPrincipal, GroupPrincipal groupPrincip
         // Note that group ownership is not used.
     }
 
+    @Override
+    public UserPrincipal lookupUserByName(Path path, String name) throws IOException {
+        if (name.endsWith("$")) {
```

Comment:
can you add a comment explaining why we are doing this here -  link to something like https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/compatibility-user-accounts-end-dollar-sign here